File: node/email-contact-form/src/cors.js
export function isOriginPermitted(req) {
  if (
    !process.env.ALLOWED_ORIGINS ||
    process.env.ALLOWED_ORIGINS === '*' ||
    !req.headers['origin']
  )
    return true;
  const allowedOriginsArray = process.env.ALLOWED_ORIGINS.split(',');
  return allowedOriginsArray.includes(req.headers['origin']);
}

export function getCorsHeaders(req) {
  if (!req.headers['origin']) return {};
  return {
    'Access-Control-Allow-Origin':
      !process.env.ALLOWED_ORIGINS || process.env.ALLOWED_ORIGINS === '*'
        ? '*'
        : req.headers['origin'],
  };
}

File: python/email_contact_form/src/cors.py
import os


def is_origin_permitted(req):
    """
    Returns true if the origin is allowed to make requests to this endpoint

    Parameters:
        req: Request object

    Returns:
        (bool): True if the origin is allowed, False otherwise
    """
    if os.getenv("ALLOWED_ORIGINS", "*") == "*":
        return True

    allowed_origins_list = os.getenv("ALLOWED_ORIGINS").split(",")
    return req.headers.get("Origin") in allowed_origins_list


def get_cors_headers(req):
    """
    Returns the CORS headers for the request

    Parameters:
        req: Request object

    Returns:
        (dict): CORS headers
    """
    if "origin" not in req.headers:
        return {}

    return {
        "Access-Control-Allow-Origin": "*"
        if os.getenv("ALLOWED_ORIGINS", "*") == "*"
        else req.headers["origin"],
    }
